packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -298,6 +342,17 @@ class BottomNavigationBar extends StatefulWidget {
     return false;
   }
 
+  // Used by the [BottomNavigation] constructor to set the `selectedLabelStyle`
+  // and `unselectedLabelStyle` parameters.
+  //
+  // If the given [TextStyle] has a non-null `fontSize`, it should be used.
+  // Otherwise, the [selectedFontSize] parameter should be used.
+  static TextStyle _defaultTextStyle(TextStyle textStyle, double fontSize) {
```

Comment:
> `return textStyle?.fontSize !=null ? textStyle : TextStyle(fontSize: fontSize);`

This doesn't quite capture it, because the user might provide a `selectedTextStyle: TextStyle(FontWeight.800)`. Here, `fontSize` is `null` but we still want to use their text style. 

> The inherit:false doesn't seem to be necessary

You're totally right, removed.

> Typically it would be effectiveSelectedTextStyle or effectiveSelectedStyle.

I could be convinced either way, I like that `effectiveSelectedTextStyle` keeps the logic in the build method (closer to the usage). Although personally I still find it cleaner to treat `selectedFontSize` as a convenience parameter used only in the constructor. That way the decision is made once and then there's a SSOT. It seems misleading to me to pass both `widget.selectedLabelStyle` and `widget.selectedFontSize` down the stack (to `_BottomNavigationTile(`, for example), when really only one of them matters. Either way the API is the same to the user.

Thoughts?

Reply:
_BottomNavigationBarState.build() could deal with rationalizing the overlapping constructor parameters and BottomNavigationBar could return to being just a dumb bag of parameters.

The rationalized styles could be passed along to _createTiles().

Reply:
OK, I create `_effectiveTextStyle(...)` inside `_BottomNavigationBarState`, then I have `_createTiles()` determine the `effectiveLabelStyle`s and pass them to `_BottomNavigationTile`.